### PR TITLE
update relaxconsts for linked vars, fix debug

### LINKED ISSUE #1201: ConstantsRelaxed only works after failed solve attempt
On convexengineering/ondemand@bf0aca9ebe222b55f9073d6e23accaef8b0996f8 I tried creating a model and using ConstantsRelaxed which originally failed, but worked after trying to solve. 

```python
In [1]: from stol import Mission

In [2]: M = Mission(sp=True)

In [3]: M.cost = M["S_{runway}"]*M[M.aircraft.topvar("W")]/M["R"]

In [5]: from gpkit.constraints.relax import ConstantsRelaxed

In [6]: cr = ConstantsRelaxed(M)

/Users/mjburton11/MIT/GPKIT/gpkit/gpkit/nomials/substitution.pyc in append_sub(sub, keys, constants, sweep, linkedsweep)
     62 
     63         if hasattr(value, "__call__") and not hasattr(value, "key"):
---> 64             linkedsweep[key] = value
     65         elif sweepsub:
     66             sweep[key] = value

TypeError: 'NoneType' object does not support item assignment

In [7]: M.localsolve("mosek")
Beginning signomial solve.

RuntimeWarning: final status of solver 'mosek' was 'UNKNOWN', not 'optimal'.

The solver's result is stored in model.program.solver_out. A result dict can be generated via program._compile_result(program.solver_out).

In [8]: cr = ConstantsRelaxed(M)

In [9]: 
```

### PR DESCRIPTION
**Pull request by bqpd**
Closes #1201

### CODE REVIEW

File: gpkit/constraints/model.py [hunk 1]
```diff
@@ -219,8 +219,8 @@ def get_relaxed(relaxvals, mapped_list, min_return=1):
     sortrelaxed = sorted(zip(relaxvals, mapped_list), key=lambda x: x[0],
                          reverse=True)
     # 0.01 is the min treshold to avoid having one at numerical precision
```
Comment (mjburton11) [inline gpkit/constraints/model.py]: Update the comment to `# 1.01`?

File: gpkit/constraints/relax.py [hunk 2]
```diff
@@ -105,8 +105,11 @@ def __init__(self, constraints, include_only=None, exclude=None):
         exclude = frozenset(exclude) if exclude else frozenset()
         include_only = frozenset(include_only) if include_only else frozenset()
         substitutions = KeyDict(constraints.substitutions)
-        constants, _, _ = parse_subs(constraints.varkeys,
-                                     constraints.substitutions)
+        constants, _, linked = parse_subs(constraints.varkeys, substitutions)
+        if linked:
+            kdc = KeyDict(constants)
+            constants.update({v: f(kdc) for v, f in linked.items()})
+        self.constants = constants
```
Comment (mjburton11) [inline gpkit/constraints/relax.py]: so was the ConstantsRelaxed not recognizing linked substitutions and that's why it was having problems creating?

Comment (bqpd) [reply gpkit/constraints/relax.py]: Yup!

File: gpkit/constraints/model.py (hunk 1 above)
Comment (bqpd) [reply gpkit/constraints/model.py]: Oops wildo

File: gpkit/constraints/relax.py (hunk 2 above)
Comment (mjburton11) [reply gpkit/constraints/relax.py]: so why didn't it have any problems after an attempted solve

Comment (bqpd) [reply gpkit/constraints/relax.py]: I think that the last value of the linked parameters might have been then present? Not the intended behaviour, now that you point it out...

Comment (mjburton11) [reply gpkit/constraints/relax.py]: uh interesting.

File: gpkit/constraints/model.py (hunk 1 above)
Comment (bqpd) [reply gpkit/constraints/model.py]: Done

File: gpkit/constraints/relax.py (hunk 2 above)
Comment (bqpd) [reply gpkit/constraints/relax.py]: created issue #1207 for that

**Comment by bqpd** [comment]
test models please

**Pull request merged**